### PR TITLE
ready hw-04

### PR DESCRIPTION
**Pull request by YaroslavKuzero**

### CODE REVIEW

File: api/auth/auth.controller.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      }
+    })
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+const logoutController = async (req, res) => {
+  try {
+    const { id } = req.user
```
Comment (odykyi) [inline api/auth/auth.controller.js:59]: what happens when user will not send Authorization header to logout endpoint ?

id will undefined 



you should throw error before code line ```const user = await User.getUserById(id)```

File: api/contacts/contacts.router.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+} = require('./contacts.controller');
+const { verifyTokenMdlw } = require('../auth/auth.checkToken')
+
+const contactsRouter = Router();
+
+// @GET /api/contacts
+contactsRouter.get('/', verifyTokenMdlw, getContactsController)
+
+// @GET /api/contacts/:contactId
+contactsRouter.get('/:contactId', getContactById)
```
Comment (odykyi) [inline api/contacts/contacts.router.js]: add verifyTokenMiddlware here

File: api/users/users.controller.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    console.error(error);
+  }
+}
+
+const updateSubscriptionController = async (req, res) => {
+  try {
+    const { subscription } = req.body
+    const { _id } = req.user;
+    const user = await User.getUserById(_id)
+    if (!user) {
```
Comment (odykyi) [inline api/users/users.controller.js:25]: you already have verified user in middleware

**Review by odykyi — CHANGES_REQUESTED** [review]
comments

File: api/contacts/contacts.router.js (hunk 2 above)
Comment (YaroslavKuzero) [reply api/contacts/contacts.router.js]: added

File: api/contacts/contacts.model.js [hunk 4]
```diff
@@ -32,19 +33,35 @@ const contactSchema = new mongoose.Schema({
...
   }
 
-  getContacts = async () => {
-    return await this.db.find();
+  getContacts = async (query) => {
+    const { page, limit, sub } = query;
+    if (sub) {
+      return await this.db.find({ subscription: sub })
+    }
+    let options = {
```
Comment (odykyi) [inline api/contacts/contacts.model.js:44]: const

**Review by odykyi — APPROVED** [review]

**Pull request merged**